consensus/poet/sgx/sawtooth_poet_sgx/poet_enclave_sgx/poet_enclave.i

```diff
@@ -344,7 +344,12 @@ def create_signup_info(originator_public_key_hash, nonce):
 
         # Check the AVR to make sure it is valid and does not indicate
         # any errors
-        _check_verification_report(verification_report, signature)
+        try:
+            _check_verification_report(verification_report, signature)
+        except ValueError as e:
+            logger.warning("Invalid attestation verification report",
```

Comment:
Isn't this a fatal error, or can we recover from this?  This would ignore it and continue to the next step which doesn't seem right.

Would it be more appropriate to propagate this up to near main() and handle the problem there (either logging as error or printing to stderr, depending on what is most appropriate) prior to exiting?

Reply:
My bad. That should have had a `return None`. 
Some of those errors could be transient and worth some retries. Something like the machine's bios being old would not be recoverable though. So then there's a choice, do we say the validator should shut down because it can't publish blocks or should it stay up extending its chain until an admin can address the bios?

Reply:
I'm concerned that the sysadmin would not see the error messages and think that the validator was working, when in fact it would be broken.  Because  of how we have logging setup, using "logger.error()" might mitigate this to some extent, since the error log is low-volume.

When I ran into this, the block publisher sits in a very very tight loop spitting out many lines of logging, so if we stay up we should probably rate limit the amount of logging generated due to this error to some level that won't fill up a smallish log directory.

The "Invalid attestation verification report" logging message is probably not good enough in isolation to indicate the state of the validator to the sysadmin (though it seems possible the str(e) makes it clearer...).  The only way I can think of to do this is setup a machine in this state and run the validator, observe the logging, then tune the logging output at the appropriate points so that what ends up in the error log is fairly obvious as to what is happening.

Reply:
I feel like if you are joining a network that is specifying sgx poet, and you think you are starting as a validator, then a fatal error is the right thing to do. Degrading to some observer behavior does not seem correct.

Reply:
I'm thinking less about startup and more about a machine that's been running for a while. Do you want the validator to halt 2 months into execution or degrade to observer? Probably still better to halt and make it strongly apparent that the machine needs servicing.

Reply:
Exiting anytime that is not immediately upon startup is completely undesirable.  No other (common) long-lived service acts that way (on purpose anyway).  There are obvious reasons not do that as well, since it will break the REST API completely, offering no opportunity to pass some sort of intelligent status back up to a monitoring service.  (This is also a reason to stay running in the first place.)

Reply:
I updated the PR to not attempt to exit on this type of error which I now throw as an environment error.
Instead it just logs as critical. Main intent of this PR now is just properly handling the "out of date" differently than a revocation condition.